### PR TITLE
Mars Ingest Watch Folder

### PR DESCRIPTION
**Pull request by foglabs**
-Adds script to check S3 location for ingest csvs, trigger ingest, and move to outbox if found
-Adds check for number of concurrent MarsIngests allowed to MarsIngestsController#create
-Adds 'completed' flag for MarsIngest to signal that all items are complete
-Adds check for MarsIngest#in_progress? in MarsIngestItemJob `after_perform` and `rescue_from` to mark ingest as completed when all items have either failed or succeeded

### CODE REVIEW

File: app/services/collection_creator.rb [hunk 1]
```diff
@@ -3,9 +3,21 @@ class CollectionCreator
   def self.find_or_create_collection(collection_name, unit_name, collection_desc, manager_user_key)
     # collection name is stored downcased in solr
     # TODO: check normalization method that fills in name_uniq_si field on Collection create
+
+    Rails.logger.info "XXXXXX PARAMS - #{collection_name} #{unit_name} #{collection_desc} #{manager_user_key}"
```
Comment (jasoncorum) [inline app/services/collection_creator.rb:7]: Does it make sense to get rid of these logging statements and assigning the Collection to the c var?

File: app/services/manifest_to_payload_mapper.rb [hunk 2]
```diff
@@ -112,6 +112,8 @@ def deep_correct_encoding(hash)
     # Returns a found (or new) collection based on the collection fields and
     # the submitter's email.
     def collection
+      Rails.logger.info "XXXXX COLLECTION NAME #{collection_hash['collection name']}"
```
Comment (jasoncorum) [inline app/services/manifest_to_payload_mapper.rb:115]: This logging too.

File: app/services/mars_ingest_watcher.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      client.delete_object({bucket: "mlavalon", key: input_key })
+
+      # tis is the (now public) url we're actually going to hit for the ingest
+      return output_key
+    end
+
+  end
+
+  def self.run_manifest(key)
+    user = User.where(email: "woo@foo.edu").first
```
Comment (jasoncorum) [inline app/services/mars_ingest_watcher.rb:69]: Is assigning woo as the user our best option here?

Comment (foglabs) [reply app/services/mars_ingest_watcher.rb:69]: I think so - the mars ingest/records need to be owned by a user, and woo has thusfar been our 'non-user user'

**Review by jasoncorum — APPROVED** [review]
⚡

**Pull request merged**